File: php/primes.php
<?php

function isPrime($num, $primes) {
    foreach($primes as $i) {
        if($num % $i == 0)
            return false;
    }

    return true;
}

$primes = array(2, 3);

for($i = 4; $i <= 100000; $i = $i + 1)
    if(isPrime($i, $primes))
        array_push($primes, $i);

?>

File: go/src/primes/primes.go
package main

const N = 1000000

func main() {
    is_prime := [N]bool{}
    is_prime[2] = true

    for i := 3; i < N; i++ {
        p := true
        for j := 2; j*j <= i && p; j++ {
            if i % j == 0 {
                p = false
            }
        }
        if p {
            is_prime[i] = true
        }
    }

    primes := make([]int, 0, 79000)
    for x := 0; x < len(is_prime)-1; x++ {
        if is_prime[x] {
            primes = append(primes, x)
        }
    }
}

